### PR TITLE
boards: fanke: Add support for FK750M1-VBT6 board

### PR DESCRIPTION
**Pull request by NU-LL**
Adds the device trees, Kconfig, and documentation files.

The following features have been confirmed working on hardware:
* LED
* SPI LCD
* UART
* W25Q64 SPI FLASH

### CODE REVIEW

File: boards/fanke/fk750m1_vbt6/fk750m1_vbt6_defconfig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Enable UART
+CONFIG_SERIAL=y
+
+# Console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable Pinctrl
+CONFIG_PINCTRL=y
+
```
Comment (nordicjm) [inline boards/fanke/fk750m1_vbt6/fk750m1_vbt6_defconfig:19]:
```suggestion
```

Comment (NU-LL) [reply boards/fanke/fk750m1_vbt6/fk750m1_vbt6_defconfig:19]: Why do we turn off PINCTRL by default here?

File: boards/fanke/fk750m1_vbt6/board.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  name: fk750m1_vbt6
+  vendor: fanke
+  socs:
+    - name: stm32h750xx
+  revision:
+    format: major.minor.patch
+    default: 1.1.0
+    revisions:
+      - name: 1.0.0
+      - name: 1.1.0
```
Comment (nordicjm) [inline boards/fanke/fk750m1_vbt6/board.yml:11]:
```suggestion
```

Comment (NU-LL) [reply boards/fanke/fk750m1_vbt6/board.yml:11]: So do we need to remove this code?

Comment (nordicjm) [reply boards/fanke/fk750m1_vbt6/board.yml:11]: you're not adding any version specific files to this PR, so yes. You can write in the documentation that it's for board version 1.1.0 or add the version files for 1.0.0 by checking the schematic differences but it doesn't make sense to have versions in board.yml that do nothing different and both build the exact same configuration

Comment (NU-LL) [reply boards/fanke/fk750m1_vbt6/board.yml:11]: I got it. Thanks a lot.

**Comment by NU-LL** [comment]
Hi @nordicjm , I've resubmitted the relevant code as requested, please help to review it again

**Review by nordicjm — DISMISSED** [review]

**Comment by kartben** [comment]
rebased and applied fixes due to recent doc infrastructure changes around board and sample descriptions.

https://github.com/zephyrproject-rtos/zephyr/compare/272b65bfec7f51c51ae59b508404c975feb62a1e..5c3ba91e78bc1874767de0de53a8782e1a6ef8f6

**Review by kartben — APPROVED** [review]

**Review by kartben — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Comment by kartben** [comment]
@NU-LL will you be able to come back to this?

File: boards/fanke/fk750m1_vbt6/fk750m1_vbt6.dts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+			colmod = <0x05>;
+			lcm = <0x2c>;
+			porch-param = [0c 0c 00 33 33];
+			cmd2en-param = [5a 69 02 01];
+			pwctrl1-param = [a4 a1];
+			pvgam-param = [D0 04 0D 11 13 2B 3F 54 4C 18 0D 0B 1F 23];
+			nvgam-param = [D0 04 0C 11 13 2C 3F 44 51 2F 1F 1F 20 23];
+			ram-param = [00 F0];
+			rgb-param = [CD 08 14];
+			mipi-mode = <MIPI_DBI_MODE_SPI_4WIRE>;
```
Comment (erwango) [inline boards/fanke/fk750m1_vbt6/fk750m1_vbt6.dts:68]: You need to rebase, to get new deifnition of `MIPI_DBI_MODE_SPI_4WIRE` which is now a string enum.
See https://github.com/zephyrproject-rtos/zephyr/pull/81293

**Review by erwango — DISMISSED** [review]

**Comment by kartben** [comment]
sorry just spotted a minor issue with the board's doc which I just fixed directly
https://github.com/zephyrproject-rtos/zephyr/compare/4c305e00964ecf440923a2330cb34608c4ed4d3d..0b4b06f12a2ede37016f1311f144f7fa1b16361c

**Review by erwango — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**